src/app.js

```diff
@@ -8,6 +8,167 @@ import './css/style.css';
 
 console.log('it loaded!');
 
+import TripList from './app/collections/trip_list';
+
+import Trip from './app/models/trip';
+import Reservation from './app/models/reservation';
+
+const tripList = new TripList();
+
+const TRIP_FIELDS = ['name', 'continent', 'about', 'category', 'weeks', 'cost'];
+
+const RESERVATION_FIELDS = ['name', 'trip_id', 'email' ];
+
+const render = function render(tripList) {
+  $('.clear').hide();
+  const tripTableElement = $('#trip-list');
+  tripTableElement.html('');
+
+  tripList.forEach((trip) => {
+    const generatedHTML = tripTemplate(trip.attributes);
+    tripTableElement.append(generatedHTML);
+    $(`#trip${trip.attributes.id}`).on('click', (event) => {
+      let tripDetails = new Trip({id: `${trip.attributes.id}`});
+      tripDetails.fetch({
+        success: renderDetails
+      });
+    });
+    $('th.sort').removeClass('current-sort-field');
+    $(`th.sort.${ tripList.comparator }`).addClass('current-sort-field');
+  });
+};
+
+const renderDetails = function renderDetails(tripDetails){
+  const $detailsElement = $('#oneTrip');
+  console.log('found detailsElement?', $detailsElement)
+  $detailsElement.html('');
+
+  const generatedHTML = detailsTemplate(tripDetails.attributes);
+  $detailsElement.append(generatedHTML);
+
+  $('#goReservationForm').on('click', function(event) {
+    renderReservationForm(event, tripDetails);
+  });
+};
+
+const renderReservationForm = function renderReservationForm(event, tripDetails){
+  $('#goReservationForm').hide();
+  const tripId = tripDetails.attributes.id
+  const $spaceReservationForm = $('#spaceReservationForm');
+  $spaceReservationForm.html('');
+  const generatedHTML = reservationTemplate({tripId: tripId});
+  // console.log(generatedHTML);
+  $spaceReservationForm.append(generatedHTML);
+  // nuevas cosas
+  event.preventDefault();
+  $('#submitForm').on('click', reserveTripHandler);
+};
+
+const  reserveTripHandler = function reserveTripHandler(event){
+  event.preventDefault();
+  console.log('inside reserveTripHandler');
+  let reservationData = {};
+
+  RESERVATION_FIELDS.forEach((field) =>{
+    const inputElement = $(`#reservation-form input[name="${ field }"]`);
+    const value = inputElement.val();
+    reservationData[field] = value;
+    inputElement.val('');
```

Comment:
Right here you're clearing all the field values, including the hidden field where you're hiding the trip id.  Instead wait to clear the values until a save is successful.  If someone fails validations, the trip Id is being cleared and thus no subsequent reservation will work.